src/views/catalog/CreateFromInstanceTypes/state/hooks/useInstanceTypeVMInitialStore.ts

```diff
@@ -0,0 +1,45 @@
+import produce from 'immer';
+import { create } from 'zustand';
+
+import { getInstanceTypeFromVolume } from '@catalog/CreateFromInstanceTypes/components/AddBootableVolumeModal/utils/utils';
+import { BootableVolume } from '@catalog/CreateFromInstanceTypes/utils/types';
+import { getBootableVolumePVCSource } from '@catalog/CreateFromInstanceTypes/utils/utils';
+
+import { instanceTypeVMStoreInitialState } from '../utils/state';
+import { InstanceTypeVMStore } from '../utils/types';
+
+import { UseBootableVolumesValues } from './useBootableVolumes';
+import { UseInstanceTypeAndPreferencesValues } from './useInstanceTypesAndPreferences';
+
+export const useInstanceTypeVMInitialStore = create<InstanceTypeVMStore>()((set, get) => {
+  return {
+    ...instanceTypeVMStoreInitialState,
+    instanceTypeDispatch: ({ type, payload }) =>
+      set(
+        produce<InstanceTypeVMStore>(({ instanceTypeVMState }) => {
+          instanceTypeVMState[type as string] = payload;
+        }),
+      ),
+    setBootableVolumesData: (data: UseBootableVolumesValues) =>
+      set((state) => ({ ...state, bootableVolumesData: data })),
+    setInstanceTypesAndPreferencesData: (data: UseInstanceTypeAndPreferencesValues) =>
+      set((state) => ({ ...state, instanceTypesAndPreferencesData: data })),
+    onSelectVolume: (selectedVolume: BootableVolume) =>
+      set(
+        produce<InstanceTypeVMStore>(({ instanceTypeVMState }) => {
+          instanceTypeVMState.selectedBootableVolume = selectedVolume;
+          instanceTypeVMState.pvcSource = getBootableVolumePVCSource(
+            selectedVolume,
+            get().bootableVolumesData.pvcSources,
+          );
+          instanceTypeVMState.selectedInstanceType = getInstanceTypeFromVolume(selectedVolume);
+        }),
+      ),
+    setActiveNamespace: (namespace: string) =>
+      set((state) => ({ ...state, activeNamespace: namespace })),
+    setVMNamespaceTarget: (namespace: string) =>
+      set((state) => ({ ...state, vmNamespaceTarget: namespace })),
+    resetInstanceTypeVMState: () =>
+      set((state) => ({ ...state, ...instanceTypeVMStoreInitialState })),
```

Comment:
why ...state ?

Reply:
you don't want to lose the setter actions :)

Reply:
you're right, zustand is nice enouge to keep previous values of unchanged fields, so I changed this file a bit to make it cleaner where I can, Thanks :)